include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -192,25 +193,19 @@ class NBL_API2 IShaderCompiler : public core::IReferenceCounted
 					{
 						assert(!_contents.empty());
 						const auto reqDirStr = requestingSourceDir.make_preferred().string();
-						std::vector<char> hashable;
-						hashable.insert(hashable.end(), reqDirStr.data()[0], reqDirStr.data()[reqDirStr.size()]);
-						hashable.insert(hashable.end(), identifier.data()[0], identifier.data()[identifier.size()]);
-						hashable.insert(hashable.end(), _contents.data()[0], _contents.data()[_contents.size()]);
+						std::vector<char> hashable(reqDirStr.size() + _identifier.size() + _contents.size());
+						hashable.insert(hashable.end(), reqDirStr.begin(), reqDirStr.end());
+						hashable.insert(hashable.end(), identifier.begin(), identifier.end());
+						hashable.insert(hashable.end(), _contents.begin(), _contents.end());
 						// Can't static cast here?
-						hash = nbl::core::XXHash_256((uint8_t*)(hashable.data()), hashable.size() * (sizeof(char) / sizeof(uint8_t)));
+						hash = nbl::core::XXHash_256(reinterpret_cast<uint8_t*>(hashable.data()), hashable.size() * (sizeof(char) / sizeof(uint8_t)));
 					}
 
-					SPreprocessingDependency(SPreprocessingDependency&) = delete;
-					SPreprocessingDependency& operator=(SPreprocessingDependency&) = delete;
-				
+					inline SPreprocessingDependency(SPreprocessingDependency&) = default;
+					inline SPreprocessingDependency& operator=(SPreprocessingDependency&) = default;
 					inline SPreprocessingDependency(SPreprocessingDependency&&) = default;
 					inline SPreprocessingDependency& operator=(SPreprocessingDependency&&) = default;
 
-					inline bool operator==(const SPreprocessingDependency& other) const
-					{
-						return hash == other.hash && identifier == identifier && contents == contents;
-					}
```

Comment:
why did you get rid of the explicit equality operator?

Reply:
It was never used - its contents are only compared against a `found_t` struct when the `find`method is called from within the Cache when trying to see if a matching entry exists

Reply:
do you use the implicit `==` at all ?

Reply:
I think not - should I delete it?

Reply:
yes for safety/performance's sake